core/src/main/scala/org/apache/spark/storage/PushBasedFetchHelper.scala

```diff
@@ -145,16 +145,23 @@ private class PushBasedFetchHelper(
         logDebug(s"Received the meta of push-merged block for ($shuffleId, $shuffleMergeId," +
           s" $reduceId) from ${req.address.host}:${req.address.port}")
         try {
-          iterator.addToResultsQueue(PushMergedRemoteMetaFetchResult(shuffleId, shuffleMergeId,
-            reduceId, sizeMap((shuffleId, reduceId)), meta.readChunkBitmaps(), address))
+          val bitmaps = meta.readChunkBitmaps()
+          if (bitmaps.nonEmpty) {
+            iterator.addToResultsQueue(PushMergedRemoteMetaFetchResult(shuffleId, shuffleMergeId,
+              reduceId, sizeMap((shuffleId, reduceId)), bitmaps, address))
+          } else {
+            logInfo(s"No available push-merged block for ($shuffleId, $shuffleMergeId," +
```

Comment:
Hmm...I thought we only raise the merged status request when there are merged blocks. Because we ensured this when fetching merge status:
https://github.com/apache/spark/blob/f1532a291179665a3b69dad640a770ecfcbed629/core/src/main/scala/org/apache/spark/MapOutputTracker.scala#L1537

But why bitmaps could be empty here? cc @mridulm @otterc